src/components/carousel/carousel.js

```diff
@@ -109,6 +190,10 @@ class Carousel extends React.Component {
     this.transitionDirection = PREVIOUS;
     this.setState({ disabled: true, selectedSlideIndex: newIndex });
     this.enableButtonsAfterTimeout();
+
+    if (this.props.onSlideChange) {
+      this.props.onSlideChange(newIndex, this.transitionDirection);
```

Comment:
These three lines could be moved into a function to remove some duplication ( although it wouldn't be much so maybe not worth it )